lib/api/xmlrpc/v1/xmlrpc.class.php

```diff
@@ -216,6 +216,10 @@ class TestlinkXMLRPCServer extends IXR_Server {
     public static $testCaseVersionParamName = "tcversion";
     public static $itsNameParamName = "itsname";
     public static $itsEnabledParamName = "itsenabled";
+    public static $itsTypeParamName = "itstype";

```

Comment:
Do not think an specific name is needed => parameter can be named only 'type'
same per enabled, same per cfg, probably same for id

Reply:
I changed some. But I'm really not sure we can always safely use attribute names without typed them. Especially some that may be used in the same time with others concepts. It may results conflicts.

Reply:
I know that in the past this approach has been the rule, but IMHO in some moments is an overkill.
Then I prefer to think about a solution if the situation will happens

regards

Reply:
If the situation happens, you will must broke the API. It's a big impact, isn't it?

Reply:
This is not going to happen if people pay attention to what is doing, testing

Reply:
I'm not sure to well understand.

Imagine I want to add a service to link a test project to an ITS. This service will take as input a test project ID and an ITS ID. Both parameters cannot have the same name ('id'). These names are used as array keys. So we must name them differently ('testprojectid' and 'itsid').
If we rename later a parameter to avoid this kind of conflict, we will break the API and will impact existing code that use it.
That's why I think it's better to distinguish parameter names without presuming the future.
If parameter names have the same names, we cannot use them together. And if we change them later, we will break existing stuff.

Reply:
nodoby have said that testprojectid parameter name has to be changed to id, but that itsid can be changed to id.
Then my choice is wait for requested changet to accept the code.

In addition: 
- sample php clients to test implementation are a good things to do
- create the method to remove link between test project and issue tracking system is IMHO needed. I've not fully read the code maybe implementation has been done passing 0 as ITS id using the set method

Reply:
> nodoby have said that testprojectid parameter name has to be changed to id, but that itsid can be changed to id.
Sure. But I think it's not a good idea to not have a global rule. In which cases I can or I cannot?...

> Then my choice is wait for requested changet to accept the code.
> 
I don't understand what do you mean.

> In addition:
> 
>     * sample php clients to test implementation are a good things to do
> 
Do you have guidelines to do this?

>     * create the method to remove link between test project and issue tracking system is IMHO needed. I've not fully read the code maybe implementation has been done passing 0 as ITS id using the set method
You're right. I will try to add it.